plugin-niassa+pinery/src/main/java/ca/on/oicr/gsi/shesmu/pinery/PinerySource.java

```diff
@@ -122,6 +122,21 @@ private ItemCache(Path fileName) {
                     lp.getSequencerRunName(), new Pair<>(runDirectory, basesMask));
                 final Instant lastModified =
                     lp.getLastModified() == null ? Instant.EPOCH : lp.getLastModified().toInstant();
+                String runStatus = "Unknown";
```

Comment:
Sadly, doing this will kill Pinery. Right now, I've got a  `Map<String, Pair<String, String>> runDirectoriesAndMasks` for the information that needs to go from lanes to samples. Could you move this outside the loop and build a `Map<String, RunDto>` and then pick form the map?

Reply:
I've made some changes, can you let me know if that's roughly what you were thinking?

Reply:
That's exactly what I was thinking. :+1:

Reply:
Documentation for 404 is added